extension/extension.go

```diff
@@ -0,0 +1,191 @@
+package extension
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+)
+
+const defaultRepoPrefix = "optimus-extension-"
+
+// Extension is manager for extension
+type Extension struct {
+	ctx context.Context
+
+	githubClient *github.Client
+	httpClient   *http.Client
+
+	manifest *Manifest
+
+	reservedCommands []string
+}
+
+// NewExtension initializes extension
+func NewExtension(
+	ctx context.Context, manifest *Manifest,
+	githubClient *github.Client, httpClient *http.Client,
+	reservedCommands ...string,
+) (*Extension, error) {
```

Comment:
I don't see this throwing error in any case?

Reply:
You're right, will remove it later.

Reply:
After some implementation changes, I decided to keep it but I added some checks which could return errors.